lightning/backend/local.go

```diff
@@ -423,16 +456,62 @@ func (local *local) Close() {
 	}
 }
 
-// Flush ensure the written data is saved successfully, to make sure no data lose after restart
-func (local *local) Flush(engineId uuid.UUID) error {
-	if engine, ok := local.engines.Load(engineId); ok {
-		engineFile := engine.(*LocalFile)
-		if err := engineFile.db.Flush(); err != nil {
-			return err
+// FlushEngine ensure the written data is saved successfully, to make sure no data lose after restart
+func (local *local) FlushEngine(engineId uuid.UUID) error {
+	engineFile, ok := local.lockEngine(engineId, importMutexStateFlush)
+
+	// the engine cannot be deleted after while we've acquired the lock identified by UUID.
+
+	if !ok {
+		return errors.Errorf("engine '%s' not found", engineId)
+	}
+	defer engineFile.unlock()
+	if err := engineFile.db.Flush(); err != nil {
+		return err
+	}
+	return local.saveEngineMeta(engineFile)
+}
+
+func (local *local) FlushAllEngines() (err error) {
+	local.lockAllEngines(importMutexStateFlush)
+	defer local.unlockAllEngines()
+
+	var flushFinishedWg sync.WaitGroup
+	var flushedEngines []*LocalFile
+
+	// first, (asynchronously) flush all opened engines
+	local.engines.Range(func(k, v interface{}) bool {
+		engineFile := v.(*LocalFile)
+		if engineFile.isLocked() {
+			// ignore engines which have been importing, flushing, or closing.
+			return true
```

Comment:
At this point
All OpenEngine is blocked due to [DiskQuotaLock](https://github.com/pingcap/tidb-lightning/pull/493/files#diff-ca1e86afc37a9b71183fb70c7fd2d65ea68a24a60e55a664d2f379be63fc7f9dR1727), so there is no new opened engine.
All exists engineFile is locked to flush state due to [L476](https://github.com/pingcap/tidb-lightning/pull/493/files#diff-3f1c53e9d349649b588b6da73533591d2f35d992bbafe5239cc3cc781f71e7e8R476), so It's always return true.
It seems the code will never reach the `AsyncFlush` logic. 🤔️